### PR TITLE
Turn menus contributions into dict of arbitrary key to list of MenuItems

### PR DESCRIPTION
**Pull request by tlambert03**
this includes #174 

@sofroniewn toyed with this pattern at some point in the hackathon... I think this is the right way for us to do menus.

Menus are still a key value pair, where the key is a string "menu id" (could be anything, not validated at the moment of manifest consumption) and the value is a List of `MenuItem`, where `MenuItem` is the union of a SubMenu or MenuCommand.

I dug into vscode a bit more, and this is how they do it.  There is no validation of the key at ingestion of the manifest.  You can put nonsense and it fails quietly.  Napari then asks for all of the contributions of a given menu key.  I don't think that "quiet" failure is a problem, particularly given that we don't know what the set of all valid keys is at that moment.  If anyone is concerned about that, I think:
1)  If we wanted to, we could provide a flag in `npe2 validate` that warns if a key is not a known napari key (i.e., if a plugin wants to opt in to _only_ ever contributing to napari keys, we can tell them if they've done anything wrong... but as a command line tool)
2) I think failing quietly is fine in this case, because a plugin developer is going to try to add something to a menu,  and if it doesn't appear, they'll go back and check it.  If you try to add something with a typo in vscode manifest, nothing shows up... and it's pretty immediately obvious during development that you made a mistake

### CODE REVIEW

File: npe2/manifest/contributions/_contributions.py [hunk 1]
```diff
@@ -20,7 +19,7 @@ class ContributionPoints(BaseModel):
     sample_data: Optional[List[SampleDataContribution]]
     themes: Optional[List[ThemeContribution]]
 
-    menus: Optional[MenusContribution] = Field(None, hide_docs=True)
+    menus: Dict[str, List[MenuItem]] = Field(default_factory=dict, description="")
```
[the remark this answers is not in the document]
Comment (nclack) [reply npe2/manifest/contributions/_contributions.py:22]: and `MenuItem`

[the remark this answers is not in the document]
Comment (tlambert03) [reply npe2/manifest/contributions/_contributions.py:23]: yep, bad merge :)

**Comment by nclack** [comment]
> There is no validation of the key at ingestion of the manifest. You can put nonsense and it fails quietly.

I think failing quietly is good. I'd like to have structural validation - that the entry in the manifest is well-formed.  But the missing foreign reference can fail quietly.

Warning on `npe2 validate` sounds good.

**Comment by tlambert03** [comment]
> I'd like to have structural validation - that the entry in the manifest is well-formed.

Yeah, all the `MenuItems` will still receive standard validation.  (i.e. it still needs to be a `List[MenuItem]`), but the menu key itself is just any string

**Pull request merged**